src/Adapter/Order/Refund/OrderRefundCalculator.php

```diff
@@ -172,18 +206,31 @@ private function isTaxIncludedInOrder(Order $order): bool
     }
 
     /**
-     * @param float $taxRate
+     * @param Order $order
      *
      * @return TaxCalculator
      *
-     * @throws PrestaShopDatabaseException
      * @throws PrestaShopException
      */
-    private function getTaxCalculator(float $taxRate)
+    private function getOrderTaxCalculator(Order $order): TaxCalculator
+    {
+        $carrier = new Carrier((int) $order->id_carrier);
+        // @todo: define if we use invoice or delivery address, or we use configuration PS_TAX_ADDRESS_TYPE
```

Comment:
PIng @eternoendless @colinegin 
I think it's mandatory before merge.
AFAIK, it's the delivery address, no?

Reply:
I don't know if it's mandatory before merging this PR, but it definitely is before the freeze so that we clean up the miscellaneous part of the code with this behaviour It is a legal/functional issue
I think that's why there is a configuration for this, so that the merchant can choose Maybe the rules are different for professional customers for example (you wan the invoice to be taxed depending on your office not where you will ship the order)

Maybe it should be related to the customer group so that you can have different settings depending on your customers (not possible for now, but it is possible to define if tax is included or not, so why not add a "tax_address_type" field?)